runtime/src/iree-amd-aie/driver/xrt/xrt_driver.cc

```diff
@@ -131,11 +110,11 @@ static iree_status_t iree_hal_xrt_driver_dump_device_info(
     iree_hal_driver_t* base_driver, iree_hal_device_id_t device_id,
     iree_string_builder_t* builder) {
   iree_hal_xrt_driver_t* driver = iree_hal_xrt_driver_cast(base_driver);
-  xrt::device* device = driver->device;
+  xrtDeviceHandle device_hdl = driver->device_hdl;
   IREE_RETURN_IF_ERROR(
       iree_string_builder_append_cstring(builder, "\n- Platform:"));
-
-  std::string platform_info = device->get_info<xrt::info::device::platform>();
+  xrt::device device(xrtDeviceToXclDevice(device_hdl));
```

Comment:
This makes exactly zero sense but it's correct - `xrtDeviceHandle` is **not** the handle that corresponds to the underlying kernel resource; the device handle that actually corresponds to the underlying kernel driver resource (the one that's deposited into the `static map<...>` mentioned below) is fetched through [this helper](https://github.com/nod-ai/XRT/blob/a01266e20dff0ec1f59059a3377102a12e2c7d89/src/runtime_src/core/common/api/xrt_device.cpp#L670).